src/core/observables/PidObservable.hpp

```diff
@@ -49,5 +53,42 @@ class PidObservable : virtual public Observable {
   std::vector<int> const &ids() const { return m_ids; }
 };
 
-} // Namespace Observables
+namespace detail {
+template <class T> struct shape_impl;
+
+template <> struct shape_impl<double> {
+  static std::vector<size_t> eval(size_t /* n_part */) { return {1}; }
```

Comment:
This size of the return type is statically known, so you could return an `array` instead of a `vector`

Reply:
yeah but shape is an overridden function and the size of the array is not the same among the observables